ns-eth/txpool.go

```diff
@@ -45,3 +68,159 @@ func (c *Client) txpoolTransaction(txHash string) {
+	/***
+	"pending": {
+		"address"/"0x00000000C0293c8cA34Dac9BCC0F953532D34e4d": {
+			"nonce"/"685615": {
+				"blockHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+				"blockNumber": null,
+				"from": "0x00000000c0293c8ca34dac9bcc0f953532d34e4d",
+				"gas": "0xfde8",
+				"gasPrice": "0x28fa6ae00",
+				"hash": "0x1ace1b9d6e834a67788a766a1674e3c28c7b816f6705db53b22c1277a4c9ffd4",
+				"input": "0xca722cdccae1eb449b0ecc06f6ad0b71b03d4eb2f7cf0d061dfd85c5d0f9f89c723d8e60c084423a922dfa3cbf0e76cf8df699b135c3d74c86360b7fcae6c618d966e176",
+				"nonce": "0xa762f",
+				"to": "0xd1ceeeeee83f8bcf3bedad437202b6154e9f5405",
+				"transactionIndex": "0x0",
+				"value": "0x0",
+				"v": "0x1c",
+				"r": "0xeacd6e6d11000a0243d17a035961621efa8364efe998ecd98392578a915739f9",
+				"s": "0x6232ba6f4f3b4648f7600bbe0b91d6c423a9950cc2e105780db9ff76fc8dca24"
+			}...
+		}...
+	}
+	"queued": {
+		"address"/"0x01Af607d196f742963200ba0259663a3E152ab3F": {
+			"nonce"/"31": {
+				"blockHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+				"blockNumber": null,
+				"from": "0x01af607d196f742963200ba0259663a3e152ab3f",
+				"gas": "0x5208",
+				"gasPrice": "0x4190ab00",
+				"hash": "0x6cbabf1c7c9638ad2f8de34d79a1c2c620e5613f0bcf48cdb3412a4d72dbf84f",
+				"input": "0x",
+				"nonce": "0x1f",
+				"to": "0xac8b30bfca3ce01bba9312571c15f4c6c8fd3a82",
+				"transactionIndex": "0x0",
+				"value": "0xbc5d37e2fc535800",
+				"v": "0x1c",
+				"r": "0xd6249d5161c3e93cd023212531d2692bdf4ecdcc3135b6d545111307f915c275",
+				"s": "0x282c36dbaf8396c9f6921ca990606d5ced440575cc10063be20e4d170a2ba5f2"
+			}...
+		}...
+	}
+	***/
+	var mempoolTx map[string]map[string]map[string]*RPCTransaction
+	err = json.Unmarshal(mp, &mempoolTx)
+	if err != nil {
+		log.Errorf("can'not unmarshal err: %v", err)
+	}
+	var mempool *sync.Map = &sync.Map{}
+	var length uint64 = 0
+	// we go on all addresses and find transaction with minimum  nonce
+	for _, addrTxs := range mempoolTx["pending"] {
+		var nonce_min uint64 = math.MaxUint64
+		var nonce_str string = ""
+		for nonce := range addrTxs {
+			nonce_uint, err := strconv.ParseUint(nonce, 10, 64)
+			if err != nil {
+				log.Errorf("Impossible to convert nonce to int. Expected value: %v, Error: %v", nonce, err)
+			}
+			if nonce_min > nonce_uint {
+				nonce_min = nonce_uint
+				nonce_str = nonce
+			}
+		}
+		var gasPrice uint64 = addrTxs[nonce_str].GasPrice.ToInt().Uint64()
+		var hash string = addrTxs[nonce_str].Hash.String()
+		mempool.Store(hash, gasPrice)
+		length++
+	}
+	log.Debugf("length adderss with tx : %v", length)
+	length = 0
+	mempool.Range(func(_, _ interface{}) bool {
+		length++
+		return true
+	})
+
+	log.Debugf("load mempool. len : %v", length)
+	c.Mempool = mempool
+	return nil
+}
+
+func (c *Client) EstimateTransactionGasPrice() *pb.GasPriceEstimation {
```

Comment:
Please return `TransactionFeeRateEstimation` here, and get rid of protobuf import